drivers/storage/port/scsiport/scsi.c

```diff
@@ -681,10 +680,24 @@ SpiProcessCompletedRequest(
     }
 
     /* Flush adapter if needed */
-    if (SrbInfo->BaseOfMapRegister)
+    if (SrbInfo->BaseOfMapRegister && SrbInfo->ScatterGather)
     {
-        /* TODO: Implement */
-        ASSERT(FALSE);
+        ULONG transferLen = 0;
+        BOOLEAN isWrite = Srb->SrbFlags & SRB_FLAGS_DATA_OUT ? TRUE : FALSE;
```

Comment:
Wouldn't the following work as well without causing any warning?
```suggestion
        BOOLEAN isWrite = Srb->SrbFlags & SRB_FLAGS_DATA_OUT;
```